lib/active_merchant/billing/gateways/realex.rb

```diff
@@ -318,7 +318,7 @@ def add_three_d_secure(xml, options)
             xml.tag! 'xid', three_d_secure[:xid]
           end
           xml.tag! 'eci', three_d_secure[:eci]
-          xml.tag! 'message_version', version
+          xml.tag! 'message_version', version.sub(/^1.*/, '1')
```

Comment:
I think that updating `version` within the branch that is specific to 1.x  helps because you do not have to question yourself about the validity of the `version.sub(/^1.*/, '1')` statement for 2.x.

Eg:
```ruby
          if /^2/.match?(version)
            xml.tag! 'authentication_value', three_d_secure[:cavv]
            xml.tag! 'ds_trans_id', three_d_secure[:ds_transaction_id]
          else
            xml.tag! 'cavv', three_d_secure[:cavv]
            xml.tag! 'xid', three_d_secure[:xid]
            version = '1'
          end
          xml.tag! 'eci', three_d_secure[:eci]
          xml.tag! 'message_version', version
```